test/unit/node_shared_test.rb

```diff
@@ -20,6 +20,32 @@ class NodeSharedTest < ActiveSupport::TestCase
     assert_equal 5, html.scan('notes-grid-test').length
     assert html.scan('<td class="title">').length > 1
   end
+ 
+  test 'that NodeShared can be used to convert short codes like [button:foo:https://google.com] into tables which list buttons' do
+    before = "Here are some notes in a table: \n\n[button:Press me:/questions] \n\n[button:Cancel:https://google.com]\n\n`[button:Cancel:https://google.com]` This shouldn't get recognized because it's in ` ticks.\n\nMake sense?"
+    html = NodeShared.button(before)
+    assert html
+    assert_equal 1, html.scan('<a class="btn btn-primary inline-button-shortcode').length
+    assert_equal 1, html.scan('href=').length
+    assert_equal 5, html.scan('inline-button-shortcode').length
```

Comment:
@jywarren I'm getting a new error (I solved the previous one - we used single quotes in the button shortcode but double in the test :) It's saying that it expects 5 but got 1 on this line (30). Did I write the test correctly (is it correct to expect 5 here?) If so, what did I write wrong in my shortcode? Thanks!

Reply:
Hmm, i think you probably just should expect 1, because we've only put one button into the sample text. The 5 was left over from the test from which you borrowed this code, i think?

Try changing it to 1, that makes sense for the test we want to do. 

You're almost there! This will be awesome! 👍 🙌

Reply:
thanks! Makes more sense :)